src/adapter/z-stack/adapter/zStackAdapter.ts

```diff
@@ -433,14 +434,23 @@ export class ZStackAdapter extends Adapter {
         disableResponse: boolean,
         disableRecovery: boolean,
         sourceEndpoint?: number,
+        profileId?: number,
     ): Promise<Events.ZclPayload | undefined> {
+        // If sourceEndpoint is not provided, try to select the source endpoint based on the profileId. If no profileId is provided, use endpoint 1.
+        const srcEndpoint = sourceEndpoint ?? Endpoints.find((e) => e.appprofid === profileId)?.endpoint ?? 1;
```

Comment:
`Endpoints.find((e) => e.appprofid === profileId)?.endpoint` should only be done when `profileId !== undefined`

Reply:
I've changed the code flow accordingly, and also moved the logic into its own function, so it can be reused for group and broadcast frames